### ISSUE TITLE
this.auth0Strategies is undefined

### ISSUE THREAD

**Issue by schwer** [auth0/lock#384]
I can reproduce the following error in firefox version 46.0.1 (though there is evidence this exists in other browsers).
With lock-9.1.4.min.js and angular-auth0-4.1.0.

To reproduce, restart firefox. Visit a page with the lock widget immediately after restarting. The issue goes away after a refresh.

I see 
![image](https://cloud.githubusercontent.com/assets/688963/15197153/e1740eb8-179d-11e6-8a0c-e67b076af764.png)

With the following error on the console:

```
TypeError: this.auth0Strategies is undefined
["/var/lib/jenkins/workspace/lock/lib/options-manager/index.js"]</</OptionsManager.prototype._getAuth0Connection()
 angular.js:33631
["/var/lib/jenkins/workspace/lock/lib/options-manager/index.js"]</</OptionsManager.prototype._isUsernameRequired()
 angular.js:33631
["/var/lib/jenkins/workspace/lock/lib/mode-signin/signin.ejs"]</anonymous/<()
 angular.js:33630
anonymous()
 angular.js:33630
["/var/lib/jenkins/workspace/lock/index.js"]</</Auth0Lock.prototype.render()
 angular.js:33627
["/var/lib/jenkins/workspace/lock/lib/mode-signin/index.js"]</SigninPanel.prototype.create()
 angular.js:33630
["/var/lib/jenkins/workspace/lock/lib/mode-signin/index.js"]</SigninPanel.prototype.render()
 angular.js:33630
["/var/lib/jenkins/workspace/lock/index.js"]</</Auth0Lock.prototype.setPanel()
 angular.js:33627
["/var/lib/jenkins/workspace/lock/index.js"]</</Auth0Lock.prototype._signinPanel()
 angular.js:33627
oninitialized()
 angular.js:33627
binded()
 angular.js:33628
finish()
 angular.js:33627
["/var/lib/jenkins/workspace/lock/index.js"]</</Auth0Lock.prototype.initialize()
 angular.js:33627
onoptionsready()
 angular.js:33627
binded()
 angular.js:33628
done()
 angular.js:33631
binded()
 angular.js:33628
g()
 angular.js:33635
["/var/lib/jenkins/workspace/lock/node_modules/browserify/node_modules/events/events.js"]</EventEmitter.prototype.emit()
 angular.js:33635
["/var/lib/jenkins/workspace/lock/lib/options-manager/index.js"]</</OptionsManager.prototype.state()
 angular.js:33631
["/var/lib/jenkins/workspace/lock/lib/options-manager/index.js"]</</OptionsManager.prototype._onclientloaded()
 angular.js:33631
binded()
 angular.js:33628
["/var/lib/jenkins/workspace/lock/index.js"]</</Auth0Lock.prototype.getClientConfiguration/<()
 angular.js:33627
g()
 angular.js:33635
["/var/lib/jenkins/workspace/lock/node_modules/browserify/node_modules/events/events.js"]</EventEmitter.prototype.emit()
 angular.js:33635
["/var/lib/jenkins/workspace/lock/index.js"]</</Auth0Lock.prototype.getClientConfiguration/global.window.Auth0.setClient()
 angular.js:33627
<anonymous>
 K5DD4oOuIPrJTKVSgYfBnDjNOdk8GzdZ.js:1
```

**Comment by hzalaz** [comment]
@schwer do you have any sample we can use to reproduce it? (or maybe a site you are using that can be accessed by us).

Also, we you logged in to the site where you see this error?

**Comment by schwer** [comment]
A little more research- I think this issue might be better suited for the auth0 angular repo. I observed a race condition when executing auth0-angular's 'auth.signin' method. The method was being invoked before the Auth0 Lock module was properly initialized.

I created a work around by listening for the 'client loaded' event on the Auth0 Lock library before executing the auth.signin method.

My previous workaround involved an arbitrary timeout, which only worked if I was lucky.
